cmake.md

```diff
@@ -0,0 +1,30 @@
+## CMake build system
+### Terminology
+
+The CMake system in symbiflow-arch-defs uses the following hierarchy:
+
+* ARCH - The architecture of FPGA.  This is used to capture architecture level parameters (e.g. tech libraries).
+* DEVICE_TYPE - A type of device.  This is defined as an ARCH and the arch.xml which defines the units with this device type.
```

Comment:
This refers to the `tile-routing-virt` and `top-routing-virt` right?

I feel like we ended up choosing a bad name for this. It would be better characterized as something like "architecture representation". Ultimately the tile and top routing should produce identical results.

For example,
 * tile routing means that a bunch of the routing is contained inside the tile definition. This is things like the local wires, global2local and similar stuff which is highly associated with the tile. Only the multi-tile spanning wires such as span4/span12/global wires are represented at the "fabric level".
 * top routing means that all the "routing" is configured to be at the "fabric level".

I guess this kind of falls apart a bit with the testarch?

Layouts might need to be special in some way, for example the `tile-routing-virt` and `top-routing-vert` have (and probably should always have) the same set of layouts?

Reply:
When you say layout, what do you mean?